src/base/RefCount.h

```diff
@@ -87,6 +87,14 @@ class RefCount
         return p.p_ != p_;
     }
 
+    bool operator == (C const * p) const {
+        return p == p_;
+    }
+
+    bool operator != (C const * p) const {
+        return p!= p_;
+    }
```

Comment:
If possible, let's support other compatible raw pointer types, as we do elsewhere in RefCount:

```suggestion
    template <class Other>
    bool operator ==(const Other * const p) const
    {
        return p == p_;
    }

    template <class Other>
    bool operator !=(const Other * const p) const
    {
        return p != p_;
    }
```

The above suggestion avoids future/hypothetical build problems like the one illustrated by the following test: https://godbolt.org/z/To4zKjv59

The above suggestion also polishes formatting and const-correctness a little.